src/Misc/virtualJoystick.ts

```diff
@@ -73,7 +91,30 @@ export class VirtualJoystick {
      * Creates a new virtual joystick
      * @param leftJoystick defines that the joystick is for left hand (false by default)
      */
-    constructor(leftJoystick?: boolean) {
+    constructor(
+        leftJoystick?: boolean,
+        customizations?: {
+            puckSize?: number,
+            containerSize?: number,
+            color?: string,
+            puckImage?: string,
+            containerImage?: string,
+            position?: { x: number, y: number },
+            alwaysVisible?: boolean,
+            limitToContainer?: boolean,
+        }
+    ) {
+        var {
+            puckSize = 40,
+            containerSize = 60,
+            color = "cyan",
+            puckImage = undefined,
+            containerImage = undefined,
+            position = undefined,
+            alwaysVisible = false,
+            limitToContainer = false,
+        } = customizations || {};

```

Comment:
we can use this kind of construct to ensure that customizations is always correctly defined:
https://github.com/BabylonJS/Babylon.js/blob/master/src/Helpers/environmentHelper.ts#L343

Reply:
@deltakosh 

Just to be clear, are you suggesting I should make an options property on virtualJoystick to hold all the customization data? Or were you just pointing out the use of an interface?

Reply:
I'm pointing the use of... :)